source/grid/grid_out.cc

```diff
@@ -3219,28 +3284,34 @@ namespace internal
                   << '\n';
             }
           else
-            // cell is at boundary and we
-            // are to treat curved
-            // boundaries. so loop over
-            // all faces and draw them as
-            // small pieces of lines
+            // cell is at boundary and we are to treat curved boundaries. so
+            // loop over all faces and draw them as small pieces of lines
             {
               for (unsigned int face_no=0;
                    face_no<GeometryInfo<dim>::faces_per_cell; ++face_no)
                 {
                   const typename dealii::Triangulation<dim,spacedim>::face_iterator
                   face = cell->face(face_no);
-                  if (face->at_boundary() || gnuplot_flags.curved_inner_cells)
+                  if (dim != spacedim || face->at_boundary() || gnuplot_flags.curved_inner_cells)
                     {
-                      // compute offset
-                      // of quadrature
-                      // points within
-                      // set of projected
-                      // points
+                      // Save the points on each face to a vector and then try
+                      // to remove colinear points that won't show up in the
+                      // generated plot.
+                      std::vector<Point<spacedim>> line_points;
+                      // compute offset of quadrature points within set of
+                      // projected points
                       const unsigned int offset=face_no*n_points;
-                      for (unsigned int i=0; i<n_points; ++i)
-                        out << (mapping->transform_unit_to_real_cell
-                                (cell, q_projector->point(offset+i)))
+                      // we don't need to transform the vertices: they are
+                      // already in physical space.
+                      line_points.push_back(face->vertex(0));
```

Comment:
I still don't understand that. Why is this so? If you were to use a `MappingQEulerian`, if you don't give the mapping to the `face->vertex()` function, how would it know the *true* location of the vertex?

Reply:
nope, you're right. Even though the documentation of this class mentions MappingQEulerian I completely missed the fact that the vertices could be moved. I'll fix this in a follow-up PR.

Reply:
OK, good I marked it up then. Thanks for offering to fix it up in a follow-up!